src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -557,19 +558,32 @@ public static ExecCredentialResponse ExecuteExternalCommand(ExternalExecution co
                 throw new KubeConfigException($"external exec failed due to: {ex.Message}");
             }
 
-            var stdout = process.StandardOutput.ReadToEnd();
-            var stderr = process.StandardError.ReadToEnd();
-            if (string.IsNullOrWhiteSpace(stderr) == false)
+
+            var stdOutTask = process.StandardOutput.ReadToEndAsync(); // Assumes process exits
+
+            var buffer = new char[4096];
+            var stdErrorTask = process.StandardError.ReadAsync(buffer, 0, buffer.Length); // Assumes process will continue
+
+            var result = Task.WaitAny(new Task[] { stdErrorTask, stdOutTask }, TimeSpan.FromSeconds(10));
```

Comment:
10 sec to config?

Reply:
what about `KubernetesClientConfiguration`

Reply:
I added a static property to KubernetesClientConfiguration.